src/plugins/intel_cpu/src/nodes/memory.hpp

```diff
@@ -152,6 +160,17 @@ class MemoryInputBase : public Input, public MemoryStateNode {
     MemoryOutputBase& getOutputNode();
     void assignState(MemStatePtr newState) override final; // NOLINT
 
+protected:
+    MemoryInputBase(const std::string id,
+                    const std::string& name,
+                    const std::string& type,
+                    const Shape& output_shape,
+                    const ov::element::Type& output_prc,
+                    const GraphContext::CPtr context,
+                    const ov::optional<std::vector<Shape>>& input_shape,
+                    const ov::optional<std::vector<ov::element::Type>>& input_prc,
+                    mode mode = mode::read_value_assign);
```

Comment:
In order to support multiple inputs, just highlight:
`const ov::optional<Shape> input_shape`
->
`const ov::optional<std::vector<Shape>> input_shape`
@maxnick